lib/core/MantraData.js

```diff
@@ -0,0 +1,303 @@
+        this.schema = relationships.find((rel) => rel.name === name);
+    };
+
+    /**
+     * Verify if the value parameter is a valid id.
+     * @param {string} value - The id to be validated.
+     * @returns {boolean} - True if the relationship name exists in the schema property.
+     */
+    _isId(value) {
+        return IS_ID_NUMBER(value);
+    };
+
+    /**
+     * Set the id property to the id parameter.
+     * Updates the path property and trimms the _path property.
+     * @param {string} id - The id value.
+     */
+    _setId(id) {
+        this.id = id;
+        this._trimPath(id);
+    };
+
+    /**
+     * Sets the id property to it's initial value.
+     */
+    _resetId() {
+        this.id = '';
+    };
+
+    /**
+     * Append the previous entity object to the parents property.
+     * @param {string} entity - The previous entity object.
+     */
+    _addParent(entity) {
+        this.parents.push(entity);
+    };
+
+    /**
+     * Sets the action property to the returned action based on the _path property.
+     */
+    setAction() {
+        this.action = this._getAction();
+    };
+
+    /**
+     * Gets a custom action or a default action object based on the _path property.
+     * @returns {object} - A default or custom action object.
+     */
+    _getAction() {
+        if (this._isCustomAction()) 
+            return this._getCustomAction();
+        else 
+            return this._getDefaultAction();
+    };
+
+    /**
+     * Verify if the first string in the _path parameter is a custom action.
+     * @returns {boolean} - True if a custom action matches the action name.
+     */
+    _isCustomAction() {
+        const [action = ''] = this._path.split('.');
+        const HAS_ACTIONS = this.schema.actions;
+        return HAS_ACTIONS && this.schema.actions.hasOwnProperty(action);
+    };
+
+    /**
+     * Gets the custom action object that matches the action name.
+     * @returns {object} - A custom action object.
+     */
+    _getCustomAction() {
+        const [action = ''] = this._path.split('.');
+        return this.schema.actions[action];
+    };
+
+    /**
+     * Verify if the default action should be the update action.
+     * @returns {boolean} - True if the id property is not empty.
+     */
+    _isUpdateAction() {
+        const IS_ID_EMPTY = (this.id !== '');
+        return IS_ID_EMPTY;
+    };
+
+    /**
+     * Gets a default action object based on MantraData's properties.
+     * @returns {object} - A built-in Mantra action object.
+     */
+    _getDefaultAction() {
+        if (this._isUpdateAction()) 
+            return DEFAULT_ACTIONS.update;
+        else 
+            return DEFAULT_ACTIONS.create;
+    };
+
+    /**
+     * Sets the component property to the returned component based on the _baseClass property.
+     */
+    setComponent() {
+        const component = this._baseClass.getComponent(this);
+        this.component = component;
```

Comment:
Let be consistent with the way we end up assigning values in void methods that its final goal is to modify a class property. There are current two alternatives being implemented at this file.
1) Create a const with the evaluation and assign the const to the property
2) Assign the property directly to the evaluation (skip the const creation) (for example at the _updatePath() method)

I suggest we take only one approach for this scenarios and I would suggest it to be the option 2).

Reply:
Sure! Will do.